components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/listeners/UserManagementAuditLogger.java

```diff
@@ -416,11 +464,31 @@ private String createAuditMessage(String action, String target, JSONObject data,
         if (data == null) {
             data = new JSONObject();
         }
+        String initiator = getInitiator();
         addContextualAuditParams(data);
         String auditMessage =
                 ListenerUtils.INITIATOR + "=%s " + ListenerUtils.ACTION + "=%s " + ListenerUtils.TARGET + "=%s "
                         + ListenerUtils.DATA + "=%s " + ListenerUtils.OUTCOME + "=%s";
-        return String.format(auditMessage, ListenerUtils.getUser(), action, target, data, resultField);
+        return String.format(auditMessage, initiator, action, target, data, resultField);
+    }
+
+    /**
+     * Get the initiator.
+     *
+     * @return the initiator based on whether log masking is enabled or not.
+     */
+    private String getInitiator() {
+
+        if (LoggerUtils.isLogMaskingEnable) {
+            String username = MultitenantUtils.getTenantAwareUsername(ListenerUtils.getUser());
+            String tenantDomain = MultitenantUtils.getTenantDomain(ListenerUtils.getUser());
+            if (StringUtils.isNotBlank(username) && StringUtils.isNotBlank(tenantDomain)) {
+                String initiator = IdentityUtil.getInitiatorId(username, tenantDomain);
+                if (StringUtils.isNotBlank(initiator)) return initiator;
```

Comment:
```suggestion
                if (StringUtils.isNotBlank(initiator)) {
                      return initiator;
                }
```

to improve the readability